runtime/worker.rs

```diff
@@ -958,3 +931,265 @@ impl MainWorker {
+    deno_kv::deno_kv::lazy_init::<MultiBackendDbHandler>(),
+    deno_cron::deno_cron::init(LocalCronHandler::new()),
+    deno_napi::deno_napi::lazy_init::<PermissionsContainer>(),
+    deno_http::deno_http::lazy_init(),
+    deno_io::deno_io::lazy_init(),
+    deno_fs::deno_fs::lazy_init::<PermissionsContainer>(),
+    deno_os::deno_os::lazy_init(),
+    deno_process::deno_process::lazy_init(),
+    deno_node::deno_node::lazy_init::<
+      PermissionsContainer,
+      TInNpmPackageChecker,
+      TNpmPackageFolderResolver,
+      TExtNodeSys,
+    >(),
+    // Ops from this crate
+    ops::runtime::deno_runtime::lazy_init(),
+    ops::worker_host::deno_worker_host::lazy_init(),
+    ops::fs_events::deno_fs_events::init(),
+    ops::permissions::deno_permissions::init(),
+    ops::tty::deno_tty::init(),
+    ops::http::deno_http_runtime::init(),
+    ops::bootstrap::deno_bootstrap::init(
+      has_snapshot.then(Default::default),
+      unconfigured,
+    ),
+    runtime::init(),
+    // NOTE(bartlomieju): this is done, just so that ops from this extension
+    // are available and importing them in `99_main.js` doesn't cause an
+    // error because they're not defined. Trying to use these ops in non-worker
+    // context will cause a panic.
+    ops::web_worker::deno_web_worker::init().disable(),
+  ]
+}
+
+#[allow(clippy::too_many_arguments)]
+fn common_runtime(
+  module_loader: Rc<dyn ModuleLoader>,
+  startup_snapshot: Option<&'static [u8]>,
+  create_params: Option<v8::CreateParams>,
+  skip_op_registration: bool,
+  shared_array_buffer_store: Option<SharedArrayBufferStore>,
+  compiled_wasm_module_store: Option<CompiledWasmModuleStore>,
+  extensions: Vec<Extension>,
+  op_metrics_factory_fn: Option<OpMetricsFactoryFn>,
+  enable_stack_trace_arg_in_ops: bool,
+) -> JsRuntime {
+  JsRuntime::new(RuntimeOptions {
+    module_loader: Some(module_loader),
+    startup_snapshot,
+    create_params,
+    skip_op_registration,
+    shared_array_buffer_store,
+    compiled_wasm_module_store,
+    extensions,
+    #[cfg(feature = "transpile")]
+    extension_transpiler: Some(Rc::new(|specifier, source| {
+      crate::transpile::maybe_transpile_source(specifier, source)
+    })),
+    #[cfg(not(feature = "transpile"))]
+    extension_transpiler: None,
+    inspector: true,
+    is_main: true,
+    op_metrics_factory_fn,
+    wait_for_inspector_disconnect_callback: Some(
+      make_wait_for_inspector_disconnect_callback(),
+    ),
+    import_meta_resolve_callback: Some(Box::new(import_meta_resolve_callback)),
+    validate_import_attributes_cb: Some(Box::new(
+      validate_import_attributes_callback,
+    )),
+    import_assertions_support: deno_core::ImportAssertionsSupport::Error,
+    maybe_op_stack_trace_callback: enable_stack_trace_arg_in_ops.then(|| {
+      Box::new(|stack| {
+        deno_permissions::prompter::set_current_stacktrace(stack)
+      }) as _
+    }),
+    ..Default::default()
+  })
+}
+
+pub struct Unconfigured {
```

Comment:
Nitpick: consider renaming to `UnconfiguredRuntime` or `PrewarmedRuntime` - I found `Unconfigured` not very intelligible when reviewing `cli/` part. Fine to do it in a follow up PR to not block this one